developer/js/tests/test-default-search-term-to-key.ts

```diff
@@ -0,0 +1,20 @@
+
+import 'mocha';
+import {assert} from 'chai';
+import { defaultWordform2Key } from '../dist/lexical-model-compiler/build-trie';
+
+
+describe('The default searchTermToKey() function', function () {
+  it('should lowercase and THEN normalize', function() {
+    // "İstanbul" has a DOTTED-I. This should be lowercased,
+    // then have its dot removed! 
+    assert.equal(defaultWordform2Key('İstanbul'), 'istanbul');
+    assert.equal(defaultWordform2Key('Diyarbakır'), 'diyarbakır');
+
+    // "skýlos" is Greek for dog 🇬🇷🐶
+    // starts with an 's' and ends with an 's'
+    // which are DIFFERENT CHARACTERS in lowercased Greek!
+    assert.equal(defaultWordform2Key('σκύλος'), 'σκυλος');
+    assert.equal(defaultWordform2Key('ΣΚΥΛΟΣ'), 'σκυλος');
```

Comment:
Eep. Second worm from the can. 🥫

```suggestion
    assert.equal(defaultWordFormToKey('σκύλος'), 'σκυλος');
    assert.equal(defaultWordFormToKey('ΣΚΥΛΟΣ'), 'σκυλος');
```

Righto. I don't think this is actually what we want, is it? Because `String.toLowerCase()` is [context-sensitive](https://tc39.es/ecma262/#sec-string.prototype.tolowercase), it does _more work_ than we want it to. The differentiation between word-final sigma and sigma is actually unhelpful.

I would think we'd be wanting a char-by-char lower case without context. I imagine we want a result more like:

```suggestion
    assert.equal(defaultWordFormToKey('σκύλος'), 'σκυλοσ');
    assert.equal(defaultWordFormToKey('ΣΚΥΛΟΣ'), 'σκυλοσ');
```

We should consider using NFKD rather than NFD. That transforms chars such as U+01C4 `Ǆ` to U+0044 U+005A `DZ` which is exactly what we want.

More discussion needed?

Ref: https://www.w3.org/International/wiki/Case_folding

Reply:
👍 for NFKD! Again, the change to the default should _only_ happen when to the compiled searchTermToKey function. I like the idea of getting rid of that context-sensitive non-sense from case folding.

I think maybe we'd want to capitalize each letter individually, then lowercase them, again, individually to deal with cases like 'ß' -> 'SS', which should be stored in key form as 'ss'.

Reply:
@mcdurdin, should I leave changing the normalization form (and adding tests for it) to a new PR, or to this PR?

Reply:
I think it belongs as a separate PR. I'm not 100% sure on the round trip on capitalisation; 'ß' -> 'SS' is the only special case I can think of for this at present but there may be others; I haven't gone and checked the Unicode specs yet...